### PR TITLE
fix: [PL-39028]: Add deployMode to helm and docker delegate

### PR DESCRIPTION
**Pull request by AnupamIO**
Add deploy mode to command for smp.

Testing:
Generated commands and applied it to bring up delegate. adding generated commands below.

SAAS
helm upgrade -i helm-delegate --namespace harness-delegate-ng --create-namespace \
  harness-delegate/harness-delegate-ng \
  --set delegateName=helm-delegate \
  --set accountId=kmpySmUISimoRrJL6NL73w \
  --set delegateToken=<> \
  --set managerEndpoint=https://localhost:9090 \
  --set delegateDockerImage=harness/delegate:latest \
  --set replicas=1 --set upgrader.enabled=false

docker run --cpus=1 --memory=2g \
  -e DELEGATE_NAME=docker-delegate \
  -e NEXT_GEN="true" \
  -e DELEGATE_TYPE="DOCKER" \
  -e ACCOUNT_ID=kmpySmUISimoRrJL6NL73w \
  -e DELEGATE_TOKEN=<> \
  -e LOG_STREAMING_SERVICE_URL=https://localhost:9090/log-service/ \
  -e MANAGER_HOST_AND_PORT=https://localhost:9090 harness/delegate:latest

SMP
helm upgrade -i helm-delegate --namespace harness-delegate-ng --create-namespace \
  harness-delegate/harness-delegate-ng \
  --set delegateName=helm-delegate \
  --set deployMode=KUBERNETES \
  --set accountId=kmpySmUISimoRrJL6NL73w \
  --set delegateToken=<> \
  --set managerEndpoint=https://localhost:9090 \
  --set delegateDockerImage=harness/delegate:latest \
  --set replicas=1 --set upgrader.enabled=false

docker run --cpus=1 --memory=2g \
  -e DELEGATE_NAME=docker-delegate \
  -e DEPLOY_MODE=KUBERNETES \
  -e NEXT_GEN="true" \
  -e DELEGATE_TYPE="DOCKER" \
  -e ACCOUNT_ID=kmpySmUISimoRrJL6NL73w \
  -e DELEGATE_TOKEN=<> \
  -e LOG_STREAMING_SERVICE_URL=https://localhost:9090/log-service/ \
  -e MANAGER_HOST_AND_PORT=https://localhost:9090 harness/delegate:latest

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48789)

### CODE REVIEW

**Comment by AnupamIO** [comment]
trigger uts

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java [hunk 1]
```diff
@@ -44,6 +49,7 @@ public class DelegateInstallationCommandServiceImpl implements DelegateInstallat
       "helm upgrade -i helm-delegate --namespace harness-delegate-ng --create-namespace \\\n"
       + "  harness-delegate/harness-delegate-ng \\\n"
       + "  --set delegateName=helm-delegate \\\n"
+      + "  --set deployMode=${deployMode} \\\n"
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:52]: can we show this deployMode option only for SMP ? Saas doesn't need it, it makes the list longer

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java [hunk 2]
```diff
@@ -29,10 +31,13 @@
 public class DelegateInstallationCommandServiceImpl implements DelegateInstallationCommandService {
   private final DelegateNgTokenService delegateNgTokenService;
   private final DelegateVersionService delegateVersionService;
+
+  private final MainConfiguration mainConfiguration;
   private static final String TERRAFORM_TEMPLATE_FLE = "/delegatetemplates/delegate-terraform-example-module.ftl";
 
   private static final String DOCKER_COMMAND = "docker run --cpus=1 --memory=2g \\\n"
       + "  -e DELEGATE_NAME=docker-delegate \\\n"
+      + "  -e DEPLOY_MODE=${deployMode} \\\n"
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:40]: same as below

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java (hunk 1 above)
Comment (AnupamIO) [reply 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java]: done

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java (hunk 2 above)
Comment (AnupamIO) [reply 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java]: done

**Comment by AnupamIO** [comment]
trigger uts

**Comment by AnupamIO** [comment]
trigger uts

**Review by bharatgoelharness — APPROVED** [review]

**Pull request merged**